contracts/pcv/curve/StableSwapOperatorV1.sol

```diff
@@ -0,0 +1,264 @@
+        _feiIndex = _3crvIndex == 0 ? 1 : 0;
+        _3crv = IStableSwap(pool).coins(_3crvIndex);
+        _3pool = _curve3pool;
+        _dai = IStableSwap(_3pool).coins(0);
+        _usdc = IStableSwap(_3pool).coins(1);
+        _usdt = IStableSwap(_3pool).coins(2);
+    }
+
+    /// @notice deposit DAI, USDC, USDT, 3crv, and FEI into the pool.
+    /// Note: the FEI has to be minted & deposited on this contract in a previous
+    ///       tx, as this contract does not use the Minter role.
+    function deposit() public override onlyPCVController whenNotPaused {
+        IFei _fei = fei();
+        uint256 _3crvVirtualPrice = IStableSwap(_3pool).get_virtual_price();
+
+        // Deposit DAI, USDC, and USDT, to get 3crv
+        uint256 _daiBalance = IERC20(_dai).balanceOf(address(this));
+        uint256 _usdcBalance = IERC20(_usdc).balanceOf(address(this));
+        uint256 _usdtBalance = IERC20(_usdt).balanceOf(address(this));
+        uint256 _3crvBalanceBefore = IERC20(_3crv).balanceOf(address(this));
+        if (_daiBalance != 0 || _usdcBalance != 0 || _usdtBalance != 0) {
+            uint256[3] memory _add3poolLiquidityAmounts;
+            _add3poolLiquidityAmounts[0] = _daiBalance;
+            _add3poolLiquidityAmounts[1] = _usdcBalance;
+            _add3poolLiquidityAmounts[2] = _usdtBalance;
+            IERC20(_dai).approve(_3pool, _daiBalance);
+            IERC20(_usdc).approve(_3pool, _usdcBalance);
+            SafeERC20.safeApprove(IERC20(_usdt), _3pool, _usdtBalance);
+            I3Pool(_3pool).add_liquidity(
+              _add3poolLiquidityAmounts,
+              0 // minimum 0 3crv lp tokens out (defensively checked below)
+            );
+        }
+
+        // get the number of 3crv held by the contract, and also
+        // check for slippage during the 3pool deposit
+        uint256 _3crvBalanceAfter = IERC20(_3crv).balanceOf(address(this));
+        uint256 _3crvBalanceFromDeposit = _3crvBalanceAfter - _3crvBalanceBefore;
+        uint256 _min3crvOut = (_daiBalance + _usdcBalance + _usdtBalance) * 1e18 / _3crvVirtualPrice * (BASIS_POINTS_GRANULARITY - depositMaxSlippageBasisPoints) / BASIS_POINTS_GRANULARITY;
+        require(_3crvBalanceFromDeposit >= _min3crvOut, "StableSwwapOperatorV1: 3pool deposit slippage too high");
+
+        // get the amount of tokens in the pool
+        (uint256 _3crvAmount, uint256 _feiAmount) = (
+            IStableSwap(pool).balances(_3crvIndex),
+            IStableSwap(pool).balances(_feiIndex)
+        );
+        // ... and the expected amount of 3crv in it after deposit
+        uint256 _3crvAmountAfter = _3crvAmount + _3crvBalanceAfter;
+
+        // get the usd value of 3crv in the pool
+        uint256 _3crvUsdValue = _3crvAmountAfter * _3crvVirtualPrice / 1e18;
+
+        // compute the number of FEI to deposit
+        uint256 _feiToDeposit = 0;
+        if (_3crvUsdValue > _feiAmount) {
+            _feiToDeposit = _3crvUsdValue - _feiAmount;
+        }
```

Comment:
why not deposit any FEI in this case? Might make more sense to just deposit proportionally